packages/dev/core/src/Gizmos/gizmo.ts

```diff
@@ -254,9 +281,14 @@ export class Gizmo implements IGizmo {
 
             // Position
             if (this.updateGizmoPositionToMatchAttachedMesh) {
-                const row = effectiveNode.getWorldMatrix().getRow(3);
-                const position = row ? row.toVector3() : new Vector3(0, 0, 0);
-                this._rootMesh.position.copyFrom(position);
+                if (this.targetAnchorPoint == GizmoAnchorPoint.Pivot && effectiveNode instanceof TransformNode) {

```

Comment:
Not 100% sure if I understand your comment correctly. If I only check if the function is there or not I would still need a typeguard or check if the return type and parameters of the function match up. What would be the benefit of that instead of validating the type with instanceof?

Reply:
faster check. instead of checking the prototype chain, it would just check for the function you need

Reply:
Ah ok understandable. To be honest I`m not super familiar with typescript. Do you mean something like following. Or would this still check against the prototype chain?
```ts
if (this.anchorPoint == GizmoAnchorPoint.Pivot && (<TransformNode>effectiveNode).getAbsolutePivotPoint) {
                    const position = (<TransformNode>effectiveNode).getAbsolutePivotPoint();
```

Reply:
yep! as simple as that